### ISSUE TITLE
message of tag error?

### ISSUE THREAD

**Issue by daidai21** [apache/rocketmq#4329]
The issue tracker is used for bug reporting purposes **ONLY** whereas feature request needs to follow the [RIP process](https://github.com/apache/rocketmq/wiki/RocketMQ-Improvement-Proposal). To avoid unnecessary duplication, please check whether there is a previous issue before filing a new one.

It is recommended to start a discussion thread in the [mailing lists](http://rocketmq.apache.org/about/contact/) in cases of discussing your deployment plan, API clarification, and other non-bug-reporting issues.
We welcome any friendly suggestions, bug fixes, collaboration, and other improvements.

Please ensure that your bug report is clear and self-contained. Otherwise, it would take additional rounds of communication, thus more time, to understand the problem itself.

Generally, fixing an issue goes through the following steps:
1. Understand the issue reported;
1. Reproduce the unexpected behavior locally;
1. Perform root cause analysis to identify the underlying problem;
1. Create test cases to cover the identified problem;
1. Work out a solution to rectify the behavior and make the newly created test cases pass;
1. Make a pull request and go through peer review;

As a result, it would be very helpful yet challenging if you could provide an isolated project reproducing your reported issue. Anyway, please ensure your issue report is informative enough for the community to pick up. At a minimum, include the following hints:

**BUG REPORT**

1. Please describe the issue you observed:

- What did you do (The steps to reproduce)?
```java
 // 实例化消息生产者Producer
        DefaultMQProducer producer = new DefaultMQProducer("filter_msg");
        // 设置NameServer的地址
        producer.setNamesrvAddr("localhost:9876");
        // 启动Producer实例
        producer.start();

        List<Message> messages = buildOrderMsgs();

        // 发送消息到一个Broker
        SendResult sendResult = producer.send(messages);
        // 通过sendResult返回消息是否成功送达
        System.out.printf("%s%n", sendResult);

        // 如果不再发送消息，关闭Producer实例。
        producer.shutdown();

private static List<Message> buildOrderMsgs() throws UnsupportedEncodingException {
        // 创建消息，并指定Topic，Tag和消息体
        String topic = "filter_msg";
        List<Message> messages = new ArrayList<>();

        // 进出口订单标tag
        messages.add(new Message(topic, "export", "OrderID001", "Hello world 0 export".getBytes(RemotingHelper.DEFAULT_CHARSET)));
        messages.add(new Message(topic, "import", "OrderID002", "Hello world 1 import".getBytes(RemotingHelper.DEFAULT_CHARSET)));
        messages.add(new Message(topic, "export", "OrderID003", "Hello world 2 export".getBytes(RemotingHelper.DEFAULT_CHARSET)));
        return messages;
    }
```

- What is expected to see?

`exportCLUSTERDefaultCluster` -> `export` 

- What did you see instead?

```java
// 实例化消费者
        DefaultMQPushConsumer consumer = new DefaultMQPushConsumer("filter_msg_");
        // 设置NameServer的地址
        consumer.setNamesrvAddr("localhost:9876");
        // 订阅Topics
        consumer.subscribe("filter_msg", "*"); // TODO: 这里tag有问题

        // 注册消息监听者
        consumer.registerMessageListener(new MessageListenerConcurrently() {
            @Override
            public ConsumeConcurrentlyStatus consumeMessage(List<MessageExt> messages, ConsumeConcurrentlyContext context) {
                for (MessageExt message : messages) {
                    System.out.println("Receive message[msgId=" + message.getMsgId() + "] "
                            + message.getTags() + " "
                            + new String(message.getBody()));
                }
                return ConsumeConcurrentlyStatus.CONSUME_SUCCESS;
            }
        });
        // 启动消费者
        consumer.start();
```

consumer log:

![image](https://user-images.githubusercontent.com/36124626/168530313-60770385-4032-4ae6-ac9c-709a81120eeb.png)

2. Please tell us about your environment:

https://github.com/daidai21/rocketmq_demo/blob/master/deploy/local/README.md

OS: mac

3. Other information (e.g. detailed explanation, logs, related issues, suggestions on how to fix, etc):

**FEATURE REQUEST**

1. Please describe the feature you are requesting.

2. Provide any additional detail on your proposed use case for this feature.

2. Indicate the importance of this issue to you (blocker, must-have, should-have, nice-to-have). Are you currently using any workarounds to address this issue?

4. If there are some sub-tasks involved, use -[] for each sub-task and create a corresponding issue to map to the sub-task:

- [sub-task1-issue-number](example_sub_issue1_link_here): sub-task1 description here, 
- [sub-task2-issue-number](example_sub_issue2_link_here): sub-task2 description here,
- ...

**Comment by ltamber** [comment]
can you tell the version of the rocketmq client & server ?

**Comment by daidai21** [comment]
4.9.1 

<dependency>
            <groupId>org.apache.rocketmq</groupId>
            <artifactId>rocketmq-client</artifactId>
            <version>4.9.1</version>
        </dependency>

**Comment by HScarb** [comment]
See #3476

**Comment by duhenglucky** [comment]
@daidai21 this bug has been fixed in this [PR](https://github.com/apache/rocketmq/issues/3486) and released in 4.9.3

**Comment by daidai21** [comment]
> @daidai21 this bug has been fixed in this [PR](https://github.com/apache/rocketmq/issues/3486) and released in 4.9.3

tks
